engine/src/engine.rs

```diff
@@ -1078,16 +1065,19 @@ pub fn submit_with_alt_modexp<
 
     check_nonce(&io, &sender, &transaction.nonce)?;
 
-    if fixed_gas_cost.is_none() {
-        // Check intrinsic gas is covered by transaction gas limit
-        match transaction.intrinsic_gas(CONFIG) {
-            Err(_e) => {
-                return Err(EngineErrorKind::GasOverflow.into());
-            }
-            Ok(intrinsic_gas) => {
-                if transaction.gas_limit < intrinsic_gas.into() {
-                    return Err(EngineErrorKind::IntrinsicGasNotMet.into());
-                }
+    // Check that fixed gas is less than gasLimit from the transaction.
+    if fixed_gas.map_or(false, |gas| gas.raw() > transaction.gas_limit) {
+        return Err(EngineErrorKind::FixedGasOverflow.into());
```

Comment:
The comment and the implementation don't match. The comment says the fixed gas should be less than the tx gas limit, but the implementation checks that the fixed gas is greater.